Core/ResponseHandler.php

```diff
@@ -82,40 +80,60 @@ private static function _getParentTransactionId(Response $oResponse)
-        $oTransaction->wdoxidee_ordertransactions__ordernumber = new Field($oOrder->oxorder__oxordernr->value);
-        $oTransaction->wdoxidee_ordertransactions__orderid = new Field($oOrder->oxorder__oxid->value);
-        $oTransaction->wdoxidee_ordertransactions__transactionid = new Field($oResponse->getTransactionId());
-        $oTransaction->wdoxidee_ordertransactions__parenttransactionid
-            = new Field(self::_getParentTransactionId($oResponse));
-        $oTransaction->wdoxidee_ordertransactions__requestid = new Field($oResponse->getRequestId());
-        $oTransaction->wdoxidee_ordertransactions__action
-            = new Field($oPayment->oxpayments__wdoxidee_transactionaction->value);
-        $oTransaction->wdoxidee_ordertransactions__type = new Field($oResponse->getTransactionType());
-        $oTransaction->wdoxidee_ordertransactions__state = new Field($aData['transaction-state']);
-
-        $oTransaction->wdoxidee_ordertransactions__amount = new Field($aData['requested-amount']);
-        $oTransaction->wdoxidee_ordertransactions__currency = new Field($aData['currency']);
-
-        $oTransaction->wdoxidee_ordertransactions__responsexml = new Field(base64_encode($oResponse->getRawData()));
-        $oTransaction->wdoxidee_ordertransactions__date = new Field(
-            Helper::getFormattedDbDate($oResponse->getData()['completion-time-stamp'])
-        );
-        //$oTransaction->wdoxidee_ordertransactions__validsignature = new Field($oResponse->isValidSignature());
-
-        $oTransaction->save();
+        $oUtilsDate = Registry::getUtilsDate();
+        $sConvertedTimestamp = $oUtilsDate->formatDBTimestamp($oUtilsDate->formTime($aData['completion-time-stamp']));
+
+        // determine if transaction state is closed
+        $sTransactionState = $oBackendService->isFinal($oResponse->getTransactionType()) ?
+            Transaction::STATE_CLOSED : Transaction::STATE_SUCCESS;
+
+        // create an array with the properties to be saved in the transaction database entry
+        $aTransactionProps = [
+            'ordernumber' => $oOrder->oxorder__oxordernr->value,
+            'orderid' => $oOrder->oxorder__oxid->value,
+            'transactionid' => $oResponse->getTransactionId(),
+            'parenttransactionid' => self::_getParentTransactionId($oResponse),
+            'requestid' => $oResponse->getRequestId(),
+            'action' => $oOrder->getOrderPayment()->oxpayments__wdoxidee_transactionaction->value,
+            'type' => $oResponse->getTransactionType(),
+            'state' => $sTransactionState,
+            'amount' => $oResponse->getRequestedAmount()->getValue(),
+            'currency' => $oResponse->getRequestedAmount()->getCurrency(),
+            'responsexml' => base64_encode($oResponse->getRawData()),
+            'date' => $sConvertedTimestamp,
+        ];
+
+        //$aTransactionProps['validsignature'] = $oResponse->isValidSignature();
```

Comment:
Please, add a todo indicating why this is commented out (although not changed with your PR):
> // TODO Enable once highlighting manipulated transactions (WDPD-120)

Reply:
👍